### PR TITLE
Adding small comms tutorial to docs

### PR DESCRIPTION
**Pull request by cjnolet**
Adding a small tutorial explaining how to use the raft comms to docs.

### CODE REVIEW

File: docs/source/using_comms.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+   #include <raft/core/device_mdspan.hpp>
+   #include <raft/util/cudart_utils.hpp>
+
+   void test_allreduce(raft::device_resources const &handle, int root) {
+     raft::comms::comms_t const& communicator = handle.get_comms();
+     cudaStream_t stream = handle.get_stream();
+     raft::device_scalar<int> temp_scalar(stream);
+
+     int to_send = 1;
+     raft::copy(temp.data(), &send, 1, stream);
```
Comment (lowener) [inline docs/source/using_comms.rst:23]:
```suggestion
     raft::copy(temp_scalar.data(), &to_send, 1, stream);
```

File: docs/source/using_comms.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+     communicator.allreduce(temp.data(), temp.data(), 1,
```
Comment (lowener) [inline docs/source/using_comms.rst:24]:
```suggestion
     communicator.allreduce(temp_scalar.data(), temp_scalar.data(), 1,
```

File: docs/source/using_comms.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+   from pylibraft.common.handle cimport device_resources
+   from cython.operator cimport dereference as deref
+
+   cdef extern from “allreduce_test.hpp”:
+       void test_allreduce(device_resources const &handle, int root) except +
+
+   def run_test_allreduce(handle, root):
+     cdef const device_resources* h = <device_resources*><size_t>handle.getHandle()
+
+   test_allreduce(deref(h), root)
```
Comment (lowener) [inline docs/source/using_comms.rst:59]: Is the indentation level wrong?

File: docs/source/using_comms.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+.. code-block:: cpp
+   :caption: Example of running test_allreduce() in MPI
+
+   #include <raft/core/mpi_comms.hpp>
+   #include <raft/core/device_resources.hpp>
+
+   raft::device_resources resource_handle;
+   // ...
+   initialize MPI_Comm
```
Comment (benfred) [inline docs/source/using_comms.rst:39]:
```suggestion
   // initialize MPI_Comm
```

File: docs/source/using_comms.rst [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+   :caption: Example of wrapping test_allreduce() w/ cython
+
+   from pylibraft.common.handle cimport device_resources
+   from cython.operator cimport dereference as deref
+
+   cdef extern from “allreduce_test.hpp”:
+       void test_allreduce(device_resources const &handle, int root) except +
+
+   def run_test_allreduce(handle, root):
+     cdef const device_resources* h = <device_resources*><size_t>handle.getHandle()
```
Comment (benfred) [inline docs/source/using_comms.rst:57]:
```suggestion
   def run_test_allreduce(handle, root):
       cdef const device_resources* h = <device_resources*><size_t>handle.getHandle()
```

File: docs/source/using_comms.rst [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+       pure=False,
+       workers=[w]
+     )
+     for w in comms.worker_addresses
+   ]
+
+   # Wait until processing is done
+   wait(dfs, timeout=5)
+
+   comms.destroy()
```
Comment (benfred) [inline docs/source/using_comms.rst:95]: Might be good to also cleanup the dask cluster: 

```suggestion
   comms.destroy()
   client.close()
   cluster.close()
```

**Review by benfred — APPROVED** [review]
lgtm!

File: docs/source/using_comms.rst (hunk 3 above)
Comment (lowener) [reply docs/source/using_comms.rst:59]:
```suggestion
     test_allreduce(deref(h), root)
```

**Pull request merged**